Cherish

memety1 opened the pull request:
super rough but I think it is a pretty good outline. IDK how to print out the equipment that is already owned so I skipped it for now because whatever I made was gonna be wrong. But the rest of it should on the right path.

memety1 committed:
Equipment selling section

supper rough but I think it is a pretty good outline. IDK how to print out the equipment that is already owned so I skipped it for now because whatever I made was gonna be wrong. But the rest of it should on the right path.

memety1 committed:
Merge branch 'Cherish' of https://github.com/Phabbits/CNIT315 into Cherish

Phabbits commented on Test.C:
```diff
@@ -183,8 +183,44 @@ int main(){
         //selling equipment
         //decrease amount in index amount if things are sold
         //increase credit if things are sold
+        //So this is a lot of pseudo code or not finished but idk how to do it fully but it a good framework 
```
Absolutely fine, a Pull Request is a good way to get feedback on your thought process.

Phabbits commented on Test.C:
```diff
@@ -183,8 +183,44 @@ int main(){
+        printf("Equipment that you own:\n");
+        //print equipment that is owned 
+        printf("Your current credits are: %d\n", player.currentCredits); //current credit
+        printf("Please select the equipment you would like to sell or select 11 to pass: "); //chose 11 becuase there are 10 options rn
```
Good comment, but this is still unnecessary hard coding. You could make this dynamically update with our code by using the constant `NUM_EQUIPMENT`.
```suggestion
        printf("Please select the equipment you would like to sell or select %d to pass: ", NUM_EQUIPMENT + 1); //chose 11 becuase there are 10 options rn
```

Phabbits replied:
You will probably want to number and print out the options before hand. Kyle is already doing this, perhaps it can be taken out of his code and put into a separate function that you each can call.

Phabbits commented on Test.C:
```diff
@@ -183,8 +183,44 @@ int main(){
+        printf("Please select the equipment you would like to sell or select 11 to pass: "); //chose 11 becuase there are 10 options rn
+        scanf("\n%d", &equipSellSelect);
+
+        while (equipSellSelect > 11) //while the user is selecting something to sell and not choseing to pass
```
You have this backwards, and avoid hard coding values.
```suggestion
        while (equipSellSelect < NUM_EQUIPMENT) //while the user is selecting something to sell and not choseing to pass
```

Phabbits commented on Test.C:
```diff
@@ -183,8 +183,44 @@ int main(){
+        {
+            printf("You have selected: ");
+            printf("%s", equipmentStock[0][0]); //not sure if that should be equipmentStock or not
+            printf("\n How many would you like to buy? ");
```
Wording.
```suggestion
            printf("\n How many would you like to sell? ");
```

Phabbits commented on Test.C:
```diff
@@ -183,8 +183,44 @@ int main(){
+            printf("%s", equipmentStock[0][0]); //not sure if that should be equipmentStock or not
+            printf("\n How many would you like to buy? ");
+            scanf("\n%d", &equipSellAmount);
+            equipSellTransaction = equipmentStock[0][2] * equipSellAmount; //not sure if that should be equipmentStock or not
```
Not a 2D array. `equipmentStock` is the correct one to use
```suggestion
            equipSellTransaction = equipmentStock[0].cost * equipSellAmount; //not sure if that should be equipmentStock or not
```

Phabbits commented on Test.C:
```diff
@@ -183,8 +183,44 @@ int main(){
+            scanf("\n%d", &equipSellAmount);
+            equipSellTransaction = equipmentStock[0][2] * equipSellAmount; //not sure if that should be equipmentStock or not
+            printf("This will sell for: %d", equipSellTransaction);
+            printf("Are you sure you want to sell? Enter 1 for Yes or 2 for No ");
```
We should probably have the option be `y` or `n` as is standard with command line interaction. Whichever option you want default, not check all the possible combinations for, you capitalize.
```suggestion
            printf("Are you sure you want to sell? y/N");
```

Phabbits commented on Test.C:
```diff
@@ -183,8 +183,44 @@ int main(){
+            printf("This will sell for: %d", equipSellTransaction);
+            printf("Are you sure you want to sell? Enter 1 for Yes or 2 for No ");
+            scanf("\n%d", &equipSellAnswer);
+            if (equipSellAnswer == 1)
```
```suggestion
            if (equipSellAnswer == "y" || equipSellAnswer == "Y")
```

Phabbits commented on Test.C:
```diff
@@ -183,8 +183,44 @@ int main(){
+                printf("Thank You!");
+                player.currentCredits = player.currentCredits + equipSellTransaction;
+                printf("Your current credits are: %d\n", player.currentCredits); //current credit
+                player.equipmentInventory[0]= player.equipmentInventory[0] - equipSellAmount; //removes equip from inventor prob wrong
```
Looks good! Will want to check to make sure they have enough inventory to sell though.

Phabbits commented on Test.C:
```diff
@@ -183,8 +183,44 @@ int main(){
+                player.equipmentInventory[0]= player.equipmentInventory[0] - equipSellAmount; //removes equip from inventor prob wrong
+            }
+            //print equipment that is owned
+            printf("Please select the equipment you would like to sell or select 11 to pass: ");
```
Avoid hardcoding
```suggestion
            printf("Please select the equipment you would like to sell or select %d to pass: ", NUM_EQUIPMENT + 1);
```

Phabbits commented on Test.C:
```diff
@@ -183,8 +183,44 @@ int main(){
+
+        while (equipSellSelect > 11) //while the user is selecting something to sell and not choseing to pass
+        {
+            printf("You have selected: ");
+            printf("%s", equipmentStock[0][0]); //not sure if that should be equipmentStock or not
```
One line of output should relate to one line of code ideally. `equipmentStock` is the correct array to use here since it stores references to all the structures with the information. `equipmentStock` is a one dimensional array with references to the `Equipment` structure, not a 2D array using [][].
```suggestion
            printf("You have selected: %s", equipmentStock[0].name); //not sure if that should be equipmentStock or not
```

Phabbits reviewed (changes requested):
Great start, not too many issues. Same thing as with Kyle, you want to avoid hardcoding values when programming. Think when we change the number of equipment, how many lines of code will I have to find and change. We will take care of code styling after this pull request, do not make style changes here.

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
new changes

Added checks before selling items to make sure they are in the players inventory and it should now display the players inventory as well

Phabbits committed:
Merge branch 'main' into Cherish

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+        
+        printf("|----------------------------- Equipment Selling ------------------------------|\n");
+        printf("Equipment that you own:\n");
+        for(int i = 0; i < NUM_EQUIPMENT; i++){
```
This does not follow the style guide, check [here](https://github.com/Phabbits/CNIT315/wiki/how-to-contribute#spaces-horizontal-spacing).
```suggestion
        for(int i=0; i<NUM_EQUIPMENT; i++){
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+        printf("Equipment that you own:\n");
+        for(int i = 0; i < NUM_EQUIPMENT; i++){
+            printf("%s %s %d %d\n", equipmentStock[i].name, equipmentStock[i].description, equipmentStock[i].cost, equipmentStock[i].messageAmount);
+            printf("Amount of %s that is in your inventory: %d", equipmentStock[i].name, player.equipmentInventory[i]);
```
Don't forget the newline character.
```suggestion
            printf("Amount of %s that is in your inventory: %d\n", equipmentStock[i].name, player.equipmentInventory[i]);
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+        scanf("\n%d", &equipSellSelect);
+
+        while (equipSellSelect < NUM_EQUIPMENT){ // while the user is selecting something to sell and not choseing to pass
+            printf("You have selected: %s", equipmentStock[0].name); // not sure if that should be equipmentStock or not
```
Newline
```suggestion
            printf("You have selected: %s\n", equipmentStock[0].name); // not sure if that should be equipmentStock or not
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+
+        while (equipSellSelect < NUM_EQUIPMENT){ // while the user is selecting something to sell and not choseing to pass
+            printf("You have selected: %s", equipmentStock[0].name); // not sure if that should be equipmentStock or not
+            printf("\n How many would you like to sell? ");
```
Per the style guide, newline go at the end in case needed on a specific hardware to actually show the line. Check [here](https://github.com/Phabbits/CNIT315/wiki/how-to-contribute#printing-output).
```suggestion
            printf("How many would you like to sell?\n");
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+        while (equipSellSelect < NUM_EQUIPMENT){ // while the user is selecting something to sell and not choseing to pass
+            printf("You have selected: %s", equipmentStock[0].name); // not sure if that should be equipmentStock or not
+            printf("\n How many would you like to sell? ");
+            scanf("\n%d", &equipSellAmount);
```
No newline characters with scanf, do them in the previous `printf `statement if you do not want the response on the same line.
```suggestion
            scanf("%d", &equipSellAmount);
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+            printf("You have selected: %s", equipmentStock[0].name); // not sure if that should be equipmentStock or not
+            printf("\n How many would you like to sell? ");
+            scanf("\n%d", &equipSellAmount);
+            equipSellTransaction = equipmentStock[0][2] * equipSellAmount; // not sure if that should be equipmentStock or not
```
`equipmentStock` is not a 2D array.
```suggestion
            equipSellTransaction = equipmentStock[0].cost * equipSellAmount; // not sure if that should be equipmentStock or not
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+            scanf("\n%d", &equipSellAmount);
+            equipSellTransaction = equipmentStock[0][2] * equipSellAmount; // not sure if that should be equipmentStock or not
+            printf("This will sell for: %d", equipSellTransaction);
+            printf("Are you sure you want to sell? y/N");
+            scanf("\n%d", &equipSellAnswer);
```
No newlines with `scanf`. I would have the response be on the same line to match the rest of the program.
```suggestion
            printf("Are you sure you want to sell? y/N ");
            scanf("%d", &equipSellAnswer);
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+            scanf("\n%d", &equipSellAnswer);
+            if (equipSellAnswer == "y" || equipSellAnswer == "Y"){ 
+                if (player.equipmentInventory[0] > 0){ // checks that the user has the item in thier inventory
+                    printf("Thank You!");
```
Newline
```suggestion
                    printf("Thank You!\n");
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+                if (player.equipmentInventory[0] > 0){ // checks that the user has the item in thier inventory
+                    printf("Thank You!");
+                    player.currentCredits = player.currentCredits + equipSellTransaction;
+                    printf("Your current credits are: %d\n", player.currentCredits); //current credit
```
I missed this comment when fixing the conflict. Comments should have one space after the `//` per the style guide.
```suggestion
                    printf("Your current credits are: %d\n", player.currentCredits); // current credit
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+                    printf("Thank You!");
+                    player.currentCredits = player.currentCredits + equipSellTransaction;
+                    printf("Your current credits are: %d\n", player.currentCredits); //current credit
+                    player.equipmentInventory[0]= player.equipmentInventory[0] - equipSellAmount; 
```
Maintain appropiate horizontal spacing per the style guide.
```suggestion
                    player.equipmentInventory[0] = player.equipmentInventory[0] - equipSellAmount; 
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+                    player.equipmentInventory[0]= player.equipmentInventory[0] - equipSellAmount; 
+                }
+                if (player.equipmentInventory == 0){ // checks that the user has the item in thier inventory
+                    printf("You do not have any of this equipment in your inventory");
+                    printf("The transcation was canceled");
```
Newlines
```suggestion
                    printf("You do not have any of this equipment in your inventory\n");
                    printf("The transcation was canceled\n");
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+                }
+            }
+            if (equipSellAnswer == "n" || equipSellAnswer == "N"){
+                printf("The transcation was canceled");
```
Newline
```suggestion
                printf("The transcation was canceled\n");
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+                printf("The transcation was canceled");
+            }
+            // print equipment that is owned
+            for(int i = 0; i < NUM_EQUIPMENT; i++){
```
Horizontal spacing per the style guide.
```suggestion
            for(int i=0; i<NUM_EQUIPMENT; i++){
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+                printf("%s %s %d %d\n", equipmentStock[i].name, equipmentStock[i].description, equipmentStock[i].cost, equipmentStock[i].messageAmount);
+                printf("Amount of %s that is in your inventory: %d", equipmentStock[i].name, player.equipmentInventory[i]);
+            }
+            printf("Please select the equipment you would like to sell or select %d to pass: ", NUM_EQUIPMENT + 1);
+            scanf("\n%d", &equipSellSelect);
```
Newline should be with `printf` and not `scanf`. Since the rest of our code has responses on the same line I am going to remove it.
```suggestion
            printf("Please select the equipment you would like to sell or select %d to pass: ", NUM_EQUIPMENT + 1);
            scanf("%d", &equipSellSelect);
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+        }
+
+        // round is complete
+        printf("|======================= This is the end of round %2d =========================|\n", currentRound);
```
We will want to add 1 to `currentRound` since it starts at 0, this is being fixed at the top by Evan.
```suggestion
        printf("|======================= This is the end of round %2d =========================|\n", currentRound + 1);
```

Phabbits reviewed (changes requested):
Awesome work! Only one logic error that I was able to find with how you referenced `equipmentStock`. The style guide suggestions may seem really picky, but there are necessary to keep everything standardized and readable. If everyone's part of their code followed a different standard it would be frustrating, and then you get the cases like with Kyle's pull request where a lot of formatting changes are made at once, and it is hard to see the actual changes.

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Update Test.C

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

memety1 committed:
Apply suggestions from code review

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+        int equipSellSelect;
+        int equipSellAmount;
+        int equipSellTransaction;
+        int equipSellAnswer;
```
You want to store characters, `y` or `n` with this.
```suggestion
        char equipSellAnswer;
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+        }
+        printf("Your current credits are: %d\n", player.currentCredits); // current credit
+        printf("Please select the equipment you would like to sell or select %d to pass: ", NUM_EQUIPMENT + 1); // chose 11 becuase there are 10 options rn
+        scanf("\n%d", &equipSellSelect);
```
No newlines in scanf
```suggestion
        scanf("%d", &equipSellSelect);
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+            equipSellTransaction = equipmentStock[0].cost * equipSellAmount; // not sure if that should be equipmentStock or not
+            printf("This will sell for: %d", equipSellTransaction);
+            printf("Are you sure you want to sell? y/N ");
+            scanf("%d", &equipSellAnswer);
```
You are trying to get a character here. The space before `%c` is to catch the "phantom" enter.
```suggestion
            scanf(" %c", &equipSellAnswer);
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+            printf("This will sell for: %d", equipSellTransaction);
+            printf("Are you sure you want to sell? y/N ");
+            scanf("%d", &equipSellAnswer);
+            if (equipSellAnswer == "y" || equipSellAnswer == "Y"){ 
```
Double quotes indicate a string, therefore you are trying to compare to a pointer here and the program won't compile.
```suggestion
            if (equipSellAnswer == 'y' || equipSellAnswer == 'Y'){ 
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+                    printf("The transcation was canceled\n");
+                }
+            }
+            if (equipSellAnswer == "n" || equipSellAnswer == "N"){
```
Same as before
```suggestion
            if (equipSellAnswer == 'n' || equipSellAnswer == 'N'){
```

Phabbits commented on Test.C:
```diff
@@ -235,6 +235,57 @@ int main(){
+            printf("How many would you like to sell?\n");
+            scanf("%d", &equipSellAmount);
+            equipSellTransaction = equipmentStock[0].cost * equipSellAmount; // not sure if that should be equipmentStock or not
+            printf("This will sell for: %d", equipSellTransaction);
```
Forgot newline
```suggestion
            printf("This will sell for: %d\n", equipSellTransaction);
```

Phabbits reviewed (changes requested):
See suggestions.

memety1 committed:
Apply suggestions from code review

Co-authored-by: Phabbits <gabrieljfrey@gmail.com>
